src/Controller/RestApiController.php

```diff
@@ -0,0 +1,180 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Www\Controller;
+
+use Www\DataObject\RestResponseDo;
+use Www\Exception\RestException;
+use YapepBase\Config;
+use YapepBase\Controller\HttpController;
+use YapepBase\Exception\ControllerException;
+use YapepBase\Mime\MimeType;
+use YapepBase\Request\IRequest;
+use YapepBase\Response\IResponse;
+use YapepBase\View\RestTemplate;
+
+abstract class RestApiController extends HttpController
+{
+    /** @var RestResponseDo */
+    protected $restResponseDo;
+
+    /** @var string */
+    private $requestBody;
+
+    /** @var array|null */
+    private $requestData;
+
+    public function __construct(IRequest $request, IResponse $response)
+    {
+        parent::__construct($request, $response);
+        $this->restResponseDo = new RestResponseDo();
+    }
+
+    public function run($action)
```

Comment:
Also as you started to declare void as return types I'd say that is missing too. But I still prefer just to omit the return type declaration in case of void. Especially that there's no such thing as void in PHP ;)